src/pudl/workspace/datastore.py

```diff
@@ -167,54 +167,46 @@ class ZenodoFetcher:
     }
 
     DOI = {
-        "sandbox": {
-            "censusdp1tract": "10.5072/zenodo.674992",
-            "eia860": "10.5072/zenodo.1222854",
-            "eia860m": "10.5072/zenodo.1225517",
-            "eia861": "10.5072/zenodo.1229930",
-            "eia923": "10.5072/zenodo.1217724",
-            "eia_bulk_elec": "10.5072/zenodo.1103572",
-            "epacamd_eia": "10.5072/zenodo.1199170",
-            "epacems": "10.5072/zenodo.672963",
-            "ferc1": "10.5072/zenodo.1070868",
-            "ferc2": "10.5072/zenodo.1188447",
-            "ferc6": "10.5072/zenodo.1098088",
-            "ferc60": "10.5072/zenodo.1098089",
-            "ferc714": "10.5072/zenodo.1098302",
-        },
-        "production": {
-            "censusdp1tract": "10.5281/zenodo.4127049",
-            "eia860": "10.5281/zenodo.8164776",
-            "eia860m": "10.5281/zenodo.8188017",
-            "eia861": "10.5281/zenodo.8231268",
-            "eia923": "10.5281/zenodo.8172818",
-            "eia_bulk_elec": "10.5281/zenodo.7067367",
-            "epacamd_eia": "10.5281/zenodo.7900974",
-            "epacems": "10.5281/zenodo.6910058",
-            "ferc1": "10.5281/zenodo.7314437",
-            "ferc2": "10.5281/zenodo.8006881",
-            "ferc6": "10.5281/zenodo.7130141",
-            "ferc60": "10.5281/zenodo.7130146",
-            "ferc714": "10.5281/zenodo.7139875",
-        },
+        # Sandbox DOIs are provided for reference
+        "censusdp1tract": "10.5281/zenodo.4127049",
+        # "censusdp1tract": "10.5072/zenodo.674992",
+        "eia860": "10.5281/zenodo.8164776",
+        # "eia860": "10.5072/zenodo.1222854",
+        "eia860m": "10.5281/zenodo.8188017",
+        # "eia860m": "10.5072/zenodo.1225517",
+        "eia861": "10.5281/zenodo.8231268",
+        # "eia861": "10.5072/zenodo.1229930",
+        "eia923": "10.5281/zenodo.8172818",
+        # "eia923": "10.5072/zenodo.1217724",
+        "eia_bulk_elec": "10.5281/zenodo.7067367",
+        # "eia_bulk_elec": "10.5072/zenodo.1103572",
+        "epacamd_eia": "10.5281/zenodo.7900974",
+        # "epacamd_eia": "10.5072/zenodo.1199170",
+        "epacems": "10.5281/zenodo.6910058",
+        # "epacems": "10.5072/zenodo.672963",
+        "ferc1": "10.5281/zenodo.7314437",
+        # "ferc1": "10.5072/zenodo.1070868",
+        "ferc2": "10.5281/zenodo.8006881",
+        # "ferc2": "10.5072/zenodo.1188447",
+        "ferc6": "10.5281/zenodo.7130141",
+        # "ferc6": "10.5072/zenodo.1098088",
+        "ferc60": "10.5281/zenodo.7130146",
+        # "ferc60": "10.5072/zenodo.1098089",
+        "ferc714": "10.5281/zenodo.7139875",
+        # "ferc714": "10.5072/zenodo.1098302",
```

Comment:
At some point I think we agree the DOIs should come out of the codebase and go into a settings file, but I'm not trying to do that in this PR.  I left the sandbox DOIs here and commented out for easy reference if someone wants to test out one of them, or look up which Zenodo archive is referenced in the sandbox.

Reply:
I wonder if turning this into pydantic settings object that could read from env variables (e.g. `PUDL_FERC1_DOI`) could be a good way to pass sandbox values here, with production defaults as... well, defaults :)

Reply:
I think we need to store the DOIs in a file in the repo (which could be used to populate env vars) so we can look them up for cache invalidation, and easily edit them eventually with PRs when new archives become available.

But for this PR I just want to get to where we can have mixed sandbox/production DOIs to make integrating new archives by hand this fall easy.

Reply:
Using a `BaseSettings` model was easy!  Now the DOIs all get validated automatically by Pydanic, and they can optionally be set using environment variables too.